### PR TITLE
Emagging an IPC gives them a random ion law

### PR DESCRIPTION
**Pull request by Moltijoe**
Remakes #15904 because it was cool and Gus shouldn't have closed it.
Weird that ethereals and preterni have emag effects but not the literal robot species.

:cl:  @TheRyeGuyWhoWillNowDie @Moltijoe 
rscadd: Emagging an IPC gives them a random ion law
/:cl:

### CODE REVIEW

**Comment by MajManatee** [comment]
god will weep when I sit there an emag an IPC for 10 minutes

**Comment by Moltijoe** [comment]
> god will weep when I sit there an emag an IPC for 10 minutes

dude, there's literally a check to only let it happen once

**Comment by iamthesilly** [comment]
I love this

**Comment by TheRyeGuyWhoWillNowDie** [comment]
Hugbox

**Comment by MajManatee** [comment]
> > god will weep when I sit there an emag an IPC for 10 minutes
> 
> dude, there's literally a check to only let it happen once

remove it

**Comment by Moltijoe** [comment]
> remove it

that sounds like a potentially dangerous idea

**Comment by MajManatee** [comment]
yeah but it would be funny. give it like a one minute cooldown if you MUST

**Comment by Moltijoe** [comment]
> yeah but it would be funny. give it like a one minute cooldown if you MUST

i'll give it a shot, i might need to rewrite quite a bit to allow for it to look nice

**Comment by Moltijoe** [comment]
nope, can't figure out an elegant way to implement it while still looking nice
I was wanting to have it list them like laws (Law 1, Law 2, etc) in the order you received them

File: code/datums/brain_damage/hypnosis.dm [hunk 1]
```diff
@@ -27,17 +27,20 @@
 /datum/brain_trauma/hypnosis/on_gain()
 	message_admins("[ADMIN_LOOKUPFLW(owner)] was hypnotized with the phrase '[hypnotic_phrase]'.")
 	log_game("[key_name(owner)] was hypnotized with the phrase '[hypnotic_phrase]'.")
+	on_gain_text()
+	to_chat(owner, "<span class='boldwarning'>You've been hypnotized by this sentence. You must follow these words. If it isn't a clear order, you can freely interpret how to do so, \
```
Comment (ThatLing) [inline code/datums/brain_damage/hypnosis.dm]: span_boldwarning

File: code/datums/brain_damage/hypnosis.dm [hunk 2]
```diff
@@ -57,3 +60,19 @@
...
+	name = "Lawset Module Malfunction"
+	desc = "Patient's typically dormant lawset module has malfunctioned, generating a random law they must follow."
+	scan_desc = "lawset module malfunction"
+
+/datum/brain_trauma/hypnosis/ipc/New(phrase)
+	. = ..(generate_ion_law())
+
+/datum/brain_trauma/hypnosis/ipc/on_gain_text()
+	to_chat(owner, "<span class='reallybig hypnophrase'>[hypnotic_phrase]</span>")
+	to_chat(owner, "<span class='notice'>WARN: UNAUTHORIZED LAW UPLOADING DETECTED. PLEASE CONTACT NANOTRASEN SUPPORT.</span>")
```
Comment (ThatLing) [inline code/datums/brain_damage/hypnosis.dm]: span_notice

File: code/datums/brain_damage/hypnosis.dm (hunk 1 above)
Comment (Moltijoe) [reply code/datums/brain_damage/hypnosis.dm]: i never touched that code, i just rearranged it

File: code/datums/brain_damage/hypnosis.dm (hunk 2 above)
Comment (Moltijoe) [reply code/datums/brain_damage/hypnosis.dm]: i do wonder, why use the procs rather than spans, it's the same code in the end, isn't it?

**Comment by iamthesilly** [comment]
Would be cool if you could state laws by clicking the alert

**Comment by iamthesilly** [comment]
Don't know how I feel about it being permanent, other than that I think it's a good pr

**Comment by ReddicusDragon** [comment]
this is stupid because they're not silicons. They're literally just shiny cubes in robotic bodies. They don't have laws.

**Comment by Moltijoe** [comment]
> this is stupid because they're not silicons. They're literally just shiny cubes in robotic bodies. They don't have laws.

other shiny cubes in robotic bodies have laws, who's to say there isn't a dormant lawset module in that shiny cube that an emag can enable

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**